testdata/config/stackdriver.yaml

```diff
@@ -55,34 +71,118 @@ spec:
+    responseSize: response.size | 0
+    requestSize: request.size | 0
+    latency: response.duration | "0ms"
   # The Monitored Resource must match a Stackdriver Monitored resource type defined at:
   #     https://cloud.google.com/monitoring/api/resources
   # Sending invalid MRs will result in the entire metric entry being rejected.
-  monitoredResourceType: "global"
+  monitoredResourceType: '"global"'
+  monitoredResourceDimensions:
+    # When using the global MR, this must be filled in, preferably matching the
+    # project_id given above in stackdriver config.
+    project_id: '"<your project ID>"'
+---
+# Note, we're trying to keep the following metric definitions in line with metrics.yaml - we add extra here to handle
+# populating monitored resource values.
+apiVersion: "config.istio.io/v1alpha2"
+kind: metric
+metadata:
+  name: stackdriver-requestcount
+  namespace: istio-config-default
+spec:
+  value: "1"
+  dimensions:
+    source: source.labels["app"] | "unknown"
+    target: target.service | "unknown"
+    service: target.labels["app"] | "unknown"
+    method: request.path | "unknown"
+    version: target.labels["version"] | "unknown"
+    response_code: response.code | 200
+  monitoredResourceType: '"global"'
+  monitoredResourceDimensions:
+    # When using the global MR, this must be filled in, preferably matching the
+    # project_id given above in stackdriver config.
+    project_id: '"<your project ID>"'
+---
+apiVersion: "config.istio.io/v1alpha2"
+kind: metric
+metadata:
+  name: stackdriver-requestduration
+  namespace: istio-config-default
+spec:
+  value: response.latency | response.duration | "0ms"
+  dimensions:
+    source: source.labels["app"] | "unknown"
+    target: target.service | "unknown"
+    service: target.labels["app"] | "unknown"
+    method: request.path | "unknown"
+    version: target.labels["version"] | "unknown"
+    response_code: response.code | 200
+  monitoredResourceType: '"global"'
+  monitoredResourceDimensions:
+    # When using the global MR, this must be filled in, preferably matching the
+    # project_id given above in stackdriver config.
+    project_id: '"<your project ID>"'
+---
+apiVersion: "config.istio.io/v1alpha2"
+kind: metric
+metadata:
+  name: stackdriver-requestsize
+  namespace: istio-config-default
+spec:
+  value: request.size | 0
+  dimensions:
+    source: source.labels["app"] | "unknown"
+    target: target.service | "unknown"
+    service: target.labels["app"] | "unknown"
+    method: request.path | "unknown"
+    version: target.labels["version"] | "unknown"
+    response_code: response.code | 200
+  monitoredResourceType: '"global"'
+  monitoredResourceDimensions:
+    # When using the global MR, this must be filled in, preferably matching the
+    # project_id given above in stackdriver config.
+    project_id: '"<your project ID>"'
+---
+apiVersion: "config.istio.io/v1alpha2"
+kind: metric
+metadata:
+  name: stackdriver-responsesize
```

Comment:
i missed the naming discussion resolution. Are we OK with dashed names ?  Should we update the other configs to have names like: `stackdriver-response-size` ?

Reply:
Yeah - dashes are fine, periods are not. I've verified this on multiple clusters. I think we should update the others too - @mandarjog thoughts?

Reply:
Talked with Kuat and removed all dashes. Apparently there's bugs in tooling that breaks dashes in some resource names.